components/cam/src/physics/rrtmgp/radiation_utils.F90

```diff
@@ -312,11 +312,15 @@ function clip_values_3d(x, min_x, max_x, varname, tolerance) result(error_messag
       end if
    end function clip_values_3d
    !-------------------------------------------------------------------------------
-   subroutine handle_error(error_message, fatal)
+   subroutine handle_error(error_message, fatal, warn)
       use cam_abortutils, only: endrun
       character(len=*), intent(in) :: error_message
       logical, intent(in), optional :: fatal
+      logical, intent(in), optional :: warn
       logical :: fatal_local = .true.
+      logical :: warn_local = .true.
```

Comment:
I think this could cause problems down the road with threaded cases. Variables that get default values set in declaration implicitly get the "save" attribute in F90, so if you were to run with multiple threads sharing this module then it's possible that the value of warn_local from a different thread gets set to `.false.`, but you assume it has a default value of `.true.`. This is why below in my previous treatment of `fatal_local`, I explicitly set a value regardless of the presence of the `fatal` argument (i.e., whether it's true or false). I probably should have deleted the extra code in the declaration of fatal_local that assign a default value to avoid confusion. Regardless, I think what you have here presents a threading bug vulnerability, so I'd suggest tweaking slightly.

Reply:
Actually, it's probably not just threaded cases. You probably change the default value to `.false.` on your first invoking of this routine with a `.false.` argument. I.e., if you set it to `.false.` in timestep 0 in the call to the temperature clipper, then ALL subsequent calls to `handle_error` probably have `warn_local = .false.`, which is not the desired behavior.

Reply:
Agreed. Fixed in recent commit.